src/hosted-buttons/index.js

```diff
@@ -20,35 +20,34 @@ export const getHostedButtonsComponent = (): HostedButtonsComponent => {
     hostedButtonId,
   }: HostedButtonsComponentProps): HostedButtonsInstance {
     const Buttons = getButtonsComponent();
-    const render = (selector) => {
+    const render = async (selector) => {
       const merchantId = getMerchantID();
+      const { html, htmlScript, style } = await getHostedButtonDetails({
```

Comment:
I've been meaning to ask why most of the async code we write uses the .then syntax instead of the async/await syntax. 

I'm all about the async/await syntax but was curious if there were any team norms around when to use either approach

Reply:
if you are working inside of `spb` its best to use `.then` and assume you are working with `ZalgoPromise`. Also if you are working in the `buttons` part of `checkout-components`

There is a very specific part of SPB that has to be synchronous or else the popup will fail. there's also a lot of code that expects zalgo promises to be returned.

If you are making new components though, its usually safe to adopt async/await and native Promises.

Reply:
2 motivations for me to use `ZalgoPromise` instead of `async/await`:
- a lot of the kraken utility functions return `ZalgoPromise` 
  - https://github.com/krakenjs/belter/blob/449034e63bca445708b62d9e08f7190a44ba4169/src/http.js#L48
  - https://github.com/krakenjs/belter/blob/449034e63bca445708b62d9e08f7190a44ba4169/src/dom.js#L216
  - https://github.com/krakenjs/post-robot/blob/30ab862e90a023531cd46d5fe8a7342c887b521b/src/drivers/receive/index.js#L147 
  - which means that:
- any consumers of those utility functions have to return a `ZalgoPromise` _or_ explicitly:
  - eslint-disable the usage of `Promise` https://github.com/krakenjs/grumbler-scripts/blob/de4b7a3490bff44cc9ebb7dc88eea87f64b96a36/packages/eslint-config-grumbler/eslintrc-browser.js#L9
  - `$FlowIssue` lines that don't use `ZalgoPromise` 

I think it's definitely worth changing the `eslint` rule to allow promises and/or rewriting (or providing alternatives to) some of the utility functions to return native promises